### ISSUE TITLE
Tags not supported

### ISSUE THREAD

**Issue by russell-sf** [jonsamwell/dart_gherkin#3]
Trying to use this with jira/xray cucumber tests but are having a problem with tags and generated reports.

Doesn't like the tag on Feature (throws exception) and tags on Scenario are not included in generated report.

Could this be added?

```
@a
Feature: One passing scenario, one failing scenario

  @b
  Scenario: Passing
    Given this step passes

  @c
  Scenario: Failing
    Given this step fails
```

Format of cucumber .feature file exported from jira/xray
```
[ #language: { language } ]
[ @{ Test Execution issue Key } ]
[ @{ Requirement issue Key } ]
Feature: [{ Requirement Issue Summary }]
    [{ Requirement Issue Description }]
     
    [Background: 
        { Pre-Conditions }]
     
    [# { Test Issue Description }]
    @{ Test Issue Key } 
    [@{ SubRequirement Issue Key 1 } ... @{ SubRequirement Issue Key N }] 
    [@{ Test Set Issue Key 1 } ... @{ Test Set Issue Key N }] 
    [@{ Test Issue Label 1 } ... @{ Test Issue Label N}]
    Scenario [Outline]: { Test Issue Summary }
        {Test Issue Steps}
```
Reference
https://relishapp.com/cucumber/cucumber/docs/formatters/json-output-formatter

**Comment by jonsamwell** [comment]
@russell-sf I have updated the JSON reporter and fixed up the tag issue.  Could you give it a try using the ```tags-modifications``` to see if it is what you are expecting.  You can use a specific branch in your pubspec file by doing the below.

```

  dart_gherkin:
    git:
      url: git://github.com/jonsamwell/dart_gherkin.git
      ref: tags-modifications
```

Let me know how you get on and I can get this merged into master and published

**Comment by russell-sf** [comment]
Hi. Thanks for the quick turnaround. I am using this along with Flutter Gherkin which won't resolve with this branch. Can you branch Flutter Gherkin as well?

**Comment by jonsamwell** [comment]
Hi @russell-sf 

I've just created a Flutter Gherkin branch which relies on this branch.  Add the below your pubspec file to test it (note that if you explicitly depend on gherkin in your pubspec not just flutter_gherkin you will need to set that to the git branch to avoid conflicts)

```
  flutter_gherkin:
    git:
      url: git://github.com/jonsamwell/flutter_gherkin.git
      ref: tag-and-json-report-changes
```

**Comment by russell-sf** [comment]
Given
```
@a
Feature: Login
  User should be able to login successfully after clicking login button.

@b @c
Scenario: User logs in successfully
    Given I expect the user enters email
    And I expect the user enters password
    When user hits Login button
    Then user should land on next screen
```

Generates this report (notice the tags)
```
[ 
   { 
      "description":"",
      "id":"login",
      "keyword":"Feature",
      "line":1,
      "name":"Login",
      "uri":".\\test_driver\\features\\login.feature",
      "tags":[ 
         { 
            "line":0,
            "name":"Tags"  <<< THIS SHOULD BE "@a"
         }
      ],
      "elements":[ 
         { 
            "keyword":"Scenario",
            "type":"scenario",
            "id":"login;user logs in successfully",
            "name":"User logs in successfully",
            "description":"",
            "line":6,
            "tags":[ 
               { 
                  "line":5,
                  "name":"b"  <<< SHOULD BE "@b"
               },
               { 
                  "line":5,
                  "name":"c"   <<< SHOULD BE "@c"
               }
            ],
            "steps":[ 
               { 
                  "keyword":"Given ",
                  "name":"I expect the user enters email",
                  "line":7,
                  "match":{ 
                     "location":".\\test_driver\\features\\login.feature:7"
                  },
                  "result":{ 
                     "status":"passed",
                     "duration":4894000000,
                     "error_message":".\\test_driver\\features\\login.feature:7\nGiven I expect the user enters email\n\nInstance of 'GherkinStepNotDefinedException'"
                  }
               }
            ]
         }
      ]
   }
]
```

**Comment by jonsamwell** [comment]
Hi @russell-sf 

Thanks for this, I have fixed the issue with the tags, the line numbers and the way exception messages are printed.

I have updated the branch so you will need to force pub to re-install the branch in your project that uses Flutter Gherkin.  The easiest way to delete the dependency and then add it again.  You will know if it works as the tags will now have the '@' prefix.

If you think it is good to go I will publish the changes.

Jon
